File: programmers 문제 모음/연습문제/최대공약수와최소공배수.py
def solution(n, m):
    num = max(n, m)
    answer = []
    for i in range(num, -1, -1):
        if (n % i == 0) and (m % i == 0):
            answer.append(i)
            answer.append(i * int(n // i) * int(m // i))
            break
    return answer


# n = 3
# m = 12
# print(solution(n, m))

File: programmers/연습문제/최대공약수와 최소공배수.kt
class Solution {
    fun solution(n: Int, m: Int): IntArray {
        var answer = intArrayOf()
        val max_num = if (n > m) n else m

        for (gcd in max_num downTo 1) {
            if (n % gcd == 0 && m % gcd == 0) {
                answer = intArrayOf(gcd, (n / gcd) * (m / gcd) * gcd)
                break
            }
        }
        return answer
    }
}